Controllers/tutorials.js

```diff
@@ -0,0 +1,39 @@
+const getAllTutorials = (req, res) => {
+  res.json({ message: `Path: ${req.path}, Method: ${req.method}` });
+};
+
+const getSingleTutorial = (req, res) => {
+  res.json({
+    message: `Path: ${req.path}, Method: ${req.method}, Id: ${req.params.id}`,
+  });
+};
+
+const createSingleTutorial = (req, res) => {
+  res.json({
+    message: `Path: ${req.path}, Method: ${req.method}, Body:  ${JSON.stringify(
+      req.body
+    )}`,
+  });
+};
+
+const updateSingleTutorial = (req, res) => {
+  res.json({
+    message: `Path: ${req.path}, Method: ${req.method}, Id: ${
+      req.params.id
+    }, Body: ${JSON.stringify(req.body)}`,
+  });
+};
```

Comment:
You're missing the corresponding http-status code for the update route you should send back to the requester. So that maybe the frontend know, that request was successful or not.
Read more [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)